src/goto-programs/string_instrumentation.cpp

```diff
@@ -792,22 +784,18 @@ void string_instrumentationt::do_strerror(
   goto_programt tmp;
 
   {
-    goto_programt::targett assignment1=tmp.add_instruction(ASSIGN);
     exprt nondet_size =
       side_effect_expr_nondett(size_type(), it->source_location);
+    tmp.add(goto_programt::make_assignment(
+      code_assignt(symbol_size.symbol_expr(), nondet_size),
```

Comment:
Just an observation: Several of the `make_X` function now just take the arguments to the `code_X` to be constructed. Should `make_assignment` also take two expressions instead of a pre-constructed `code_assignt`?

Reply:
Yes, will add these.
Another option is to offer `goto_programt::make(code_assignt())`. The key deficiency is that this doesn't work for all instructions, and thus, creates asymmetry.

Reply:
And, of course, another option is to use initializer lists. Say you can write
`goto_programt::make_assignment({ function_call.lhs(), rhs }, some_location)`
Any preferences?

Reply:
Late response to that one: I'd only use initializer lists when we can forward them to a constructor of the to-be-build `code_*t`. Otherwise we're stuck with a list that we need to take apart in order to construct the object.

Reply:
I don't believe these are actually lists; it's syntax for the constructor call, i.e., I believe you'd get the same machine code as for `goto_programt::make_assignment(code_assignt( lhs, rhs ), some_location)`. Thus, this is a question of which you consider better for readability.

Reply:
Ah, I was thinking to hide completely the fact that we construct a `code_assignt` here. Maybe one day we don't even want/need that anymore. As a caller, I want to say "make a an assignment of `rhs` to `lhs` at `some_location`" and as the caller I really don't want to care how that's stored.

Reply:
Ok, let's go for that. I'll do the PR.